dll/win32/msgina/shutdown.c

```diff
@@ -848,140 +1057,170 @@ ShutdownOnInit(
-        idx = SendMessageW(hwndList, CB_ADDSTRING, 0, (LPARAM)szBuffer);
-        if (idx != CB_ERR)
-            SendMessageW(hwndList, CB_SETITEMDATA, idx, WLX_SAS_ACTION_SHUTDOWN_POWER_OFF);
-    }
-    else if (pContext->bFriendlyUI)
-    {
-        EnableWindow(GetDlgItem(hDlg, IDC_BUTTON_SHUTDOWN), FALSE);
-    }
+        /* Clear the content before it's used */
+        SendMessageW(hwndList, CB_RESETCONTENT, 0, 0);
 
-    /* Restart */
-    if (pContext->ShutdownOptions & WLX_SHUTDOWN_STATE_REBOOT)
-    {
-        LoadStringW(pgContext->hDllInstance, IDS_SHUTDOWN_RESTART, szBuffer, _countof(szBuffer));
-        idx = SendMessageW(hwndList, CB_ADDSTRING, 0, (LPARAM)szBuffer);
-        if (idx != CB_ERR)
-            SendMessageW(hwndList, CB_SETITEMDATA, idx, WLX_SAS_ACTION_SHUTDOWN_REBOOT);
-    }
-    else if (pContext->bFriendlyUI)
-    {
-        EnableWindow(GetDlgItem(hDlg, IDC_BUTTON_REBOOT), FALSE);
-    }
+        /* Log off */
+        if (pContext->ShutdownOptions & WLX_SHUTDOWN_STATE_LOGOFF)
+        {
+            LoadStringW(pgContext->hDllInstance, IDS_SHUTDOWN_LOGOFF, szBuffer, _countof(szBuffer));
+            StringCchPrintfW(szBuffer2, _countof(szBuffer2), szBuffer, pgContext->UserName);
+            idx = SendMessageW(hwndList, CB_ADDSTRING, 0, (LPARAM)szBuffer2);
+            if (idx != CB_ERR)
+                SendMessageW(hwndList, CB_SETITEMDATA, idx, WLX_SAS_ACTION_LOGOFF);
+        }
 
-    // if (pContext->ShutdownOptions & 0x08) {}
+        /* Shut down - DEFAULT */
+        if (pContext->ShutdownOptions & WLX_SHUTDOWN_STATE_POWER_OFF)
+        {
+            LoadStringW(pgContext->hDllInstance, IDS_SHUTDOWN_SHUTDOWN, szBuffer, _countof(szBuffer));
+            idx = SendMessageW(hwndList, CB_ADDSTRING, 0, (LPARAM)szBuffer);
+            if (idx != CB_ERR)
+                SendMessageW(hwndList, CB_SETITEMDATA, idx, WLX_SAS_ACTION_SHUTDOWN_POWER_OFF);
+        }
 
-    /* Sleep */
-    if (pContext->ShutdownOptions & WLX_SHUTDOWN_STATE_SLEEP)
-    {
-        LoadStringW(pgContext->hDllInstance, IDS_SHUTDOWN_SLEEP, szBuffer, _countof(szBuffer));
-        idx = SendMessageW(hwndList, CB_ADDSTRING, 0, (LPARAM)szBuffer);
-        if (idx != CB_ERR)
-            SendMessageW(hwndList, CB_SETITEMDATA, idx, WLX_SAS_ACTION_SHUTDOWN_SLEEP);
-    }
-    else if (pContext->bFriendlyUI)
-    {
-        EnableWindow(GetDlgItem(hDlg, IDC_BUTTON_SLEEP), IsPwrSuspendAllowed());
-    }
+        /* Restart */
+        if (pContext->ShutdownOptions & WLX_SHUTDOWN_STATE_REBOOT)
+        {
+            LoadStringW(pgContext->hDllInstance, IDS_SHUTDOWN_RESTART, szBuffer, _countof(szBuffer));
+            idx = SendMessageW(hwndList, CB_ADDSTRING, 0, (LPARAM)szBuffer);
+            if (idx != CB_ERR)
+                SendMessageW(hwndList, CB_SETITEMDATA, idx, WLX_SAS_ACTION_SHUTDOWN_REBOOT);
+        }
 
-    // if (pContext->ShutdownOptions & 0x20) {}
+        // if (pContext->ShutdownOptions & 0x08) {}
```

Comment:
What are these comments doing here?

Reply:
@learn-more These were present in the function originally so I decided to keep it.

Reply:
please don't.